### PR TITLE
Benchmark `request-vc` cli command

### PR DESCRIPTION
**Pull request by kziemianek**
Simple CLI command to run benchmark of request-vc command.

### CODE REVIEW

File: tee-worker/cli/src/benchmark/direct_request_vc.rs
Comment (Kailai-Wang) [inline tee-worker/cli/src/benchmark/direct_request_vc.rs:1]: Shall we remove the `direct` word (same below and in code)?
We will only have one way to request-vc: via native call.

See P-963

File: tee-worker/cli/src/benchmark/direct_request_vc.rs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+};
+use litentry_primitives::{RequestAesKey, ShardIdentifier, REQUEST_AES_KEY_LEN};
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
+use sgx_crypto_helper::rsa3072::Rsa3072PubKey;
+use sp_core::Pair;
+use std::time::Instant;
+
+// Command to perform direct request_vc benchmarking. It creates a given number of threads and sends
+// `requests_per_thread` requests one by one recording time to get response. It uses Alice as
+// an IdGraph root. If bench case requires any identities to be linked, the preparation should be done
```
Comment (Kailai-Wang) [inline tee-worker/cli/src/benchmark/direct_request_vc.rs]: Does it mean that the IDGraph can only be keyed by Alice? Like I have to link my identities to Alice and then request-vc under the name of Alice?

While this should be fine too, can we actually request vc with real IDGraph without Alice? Alice is used for signing only

File: tee-worker/cli/src/benchmark/direct_request_vc.rs [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+		let threads = self.threads;
+
+		let mut clients = Vec::with_capacity(threads as usize);
+
+		let assertion = self.assertion.to_assertion().unwrap();
+		let aes_key = random_aes_key();
+
+		let top = TrustedCall::request_vc(
+			alice.public().into(),
+			alice.public().into(),
```
Comment (Kailai-Wang) [inline tee-worker/cli/src/benchmark/direct_request_vc.rs]: I mean here uses a real IDGraph owner

**Review by Kailai-Wang — APPROVED** [review]
Looks good - I'm wondering what's the time for EVM assertion :D

File: tee-worker/cli/src/benchmark/direct_request_vc.rs (hunk 1 above)
Comment (kziemianek) [reply tee-worker/cli/src/benchmark/direct_request_vc.rs]: I've omitted account to whom the vc will be issued in order to simplify usage.

Yes, we can read it from cli args in order to have better flexibility.

**Comment by kziemianek** [comment]
> Looks good - I'm wondering what's the time for EVM assertion :D

Looks promising, times are almost equal for `A1`


```
kziemianek@kziemianek:~/projects/litentry/litentry-parachain/tee-worker/target/release$ ./litentry-cli trusted benchmark-request-vc 8 400 did:litentry:substrate:0x8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48 a1
Running request_vc benchmarking
Total time: 18.0592s
Average request vc time is: 0.0451s
Errors count: 0
```
```
kziemianek@kziemianek:~/projects/litentry/litentry-parachain/tee-worker/target/release$ ./litentry-cli trusted benchmark-request-vc 8 400       did:litentry:substrate:0x8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48       dynamic 0000000000000000000000000000000000000000
Running request_vc benchmarking
Total time: 18.1130s
Average request vc time is: 0.0451s
Errors count: 0
```

File: tee-worker/cli/src/benchmark/direct_request_vc.rs (hunk 2 above)
Comment (kziemianek) [reply tee-worker/cli/src/benchmark/direct_request_vc.rs]: done

**Comment by wli-pro** [comment]
The benchmark LGTM
I am also wondering the result diff for token-holding-amount VC, as the time-consuming operation mostly caused by third-party API calls

**Comment by kziemianek** [comment]
@higherordertech here are results for BNB token holding with mocked nodereal instance:


```
Preparation:

./litentry-cli trusted -d link-identity did:litentry:substrate:0x8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48 did:litentry:evm:0x0D9bFD1f18f5f4FD08247DC54aD3528909c4b3E9 bsc,ethereum
./litentry-cli trusted -d link-identity did:litentry:substrate:0x8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48 did:litentry:evm:0x0D9bFD1f18f5f4FD08247DC54aD3528909c4b3EA bsc,ethereum


Results:

kziemianek@kziemianek:~/projects/litentry/litentry-parachain/tee-worker/target/release$ ./litentry-cli trusted benchmark-request-vc 8 400 did:litentry:substrate:0x8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48 token-
holding-amount bnb
Running request_vc benchmarking
Total time: 18.1449s
Average request vc time is: 0.0452s
Errors count: 0
kziemianek@kziemianek:~/projects/litentry/litentry-parachain/tee-worker/target/release$ ./litentry-cli trusted benchmark-request-vc 8 400 did:litentry:substrate:0x8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48 dynamic 0000000000000000000000000000000000000000 00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000003626e620000000000000000000000000000000000000000000000000000000000
Running request_vc benchmarking
Total time: 18.3935s
Average request vc time is: 0.0455s
Errors count: 0
```

**Pull request merged**